Hide import sections when no file present

Atlas139mkm opened the pull request:
<!--

### Requirements

* Filling out the template is required. Any pull request that does not include
enough information to be reviewed in a timely manner may be closed at the
maintainers' discretion.
* Follow the check list items defined by https://github.com/constellation-app/constellation/blob/master/CONTRIBUTING.md#pull-requests
* All new code requires unit tests to ensure they work as expected and will
continue to work as new code is added in the future (regression testing).
* Make sure your branch name is prefixed by `feature`, `bugfix`, `hotfix` or `release`
* Have you read Constellation's Code of Conduct? By filing an issue, you are
expected to comply with it, including treating everyone with respect:
https://github.com/constellation-app/constellation/blob/master/CODE_OF_CONDUCT.md

-->

### Description of the Change

In the import view I have collapsed the configuration section and disabled the import button by default. These will be expanded and enabled respectively when there is a file present within the imported files list.

- The user can choose to expand the configuration section without importing a (nice for developers)
    - however the import button will stay disabled while no file is present
-  When the user imports a file after manually expanding the configuration section, it will stay open and the import button will enable
- a demonstration of the changes is shown in the gif below

![disabledImportView](https://user-images.githubusercontent.com/88649439/132145230-6b17f5c6-4be5-4b23-a56a-e0691a0e6dc1.gif)

<!--

We must be able to understand the design of your change from this description.
If we can't get a good idea of what the code will be doing from the description
here, the pull request may be closed at the maintainers' discretion. Keep in
mind that the maintainer reviewing this PR may not be familiar with or have
worked with the code here recently, so please walk us through the concepts.

-->

### Alternate Designs

- we could completely disable the configuration section if requested (let me know and ill see what I can do)

<!--

Explain what other alternates were considered and why the proposed version was
selected.

-->

### Why Should This Be In Core?

It assists the user in the desired flow of the view. Guiding them to import a file first before attempting to assign attributes and import it to the graph.

<!--

Explain why this functionality should be in Constellation Core as opposed to a
different module suite. Note that this question is more applicable when adding
new functionality. If this change is a minor update to an existing file then it
is understood that this change has to be to this module suite and a response
and therefore a response to this question is not required.

-->

### Benefits

UX enhancements

<!-- What benefits will be realized by the code change? -->

### Possible Drawbacks

the final import button is disabled without first importing a file. (there might be a use case where this is wanted but likely not)

<!-- What are the possible side-effects or negative impacts of the code change? -->

### Verification Process

- ensured the config pane expanded on file import
- ensured the import button enabled on file import
- ensured they both stayed active after importing 2 files
- ensured the config pane remained expanded after manually expanding it then importing a file
- ensured both config pane and import button disabled when no more files were in the import list

<!--

What process did you follow to verify that your change has the desired effects?

- How did you verify that all new functionality works as expected?
- How did you verify that all changed functionality works as expected?
- How did you verify that the change has not introduced any regressions?

Describe the actions you performed (e.g. buttons you clicked, text you typed,
commands you ran, etc.), and describe the results you observed.

-->

### Applicable Issues

#1307

arcturus2 reviewed (changes requested):
A really nice implementation @Atlas139mkm and it works well. Can you please update `whatsnew.txt` to advertise this new feature please?

antares1470 commented on CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/ActionPane.java:
```diff
@@ -77,4 +79,9 @@ private void displayAlert(final String title, final String header, final boolean
     }
+
+    // sets the enabled state of the import button
+    public void enableButton(boolean isEnabled) {
```
parameter final

antares1470 commented on CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/ImportPane.java:
```diff
@@ -200,4 +203,12 @@ public SourcePane getSourcePane() {
         return this.getScene().getWindow();
     }
+
+    public void expandPane(boolean isExpanded) {
+        titledConfigurationPane.setExpanded(isExpanded);
+    }
+
+    public void enableButton(boolean isEnabled) {
```
parameters final

antares1470 commented on CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/DelimitedImportController.java:
```diff
@@ -219,4 +222,16 @@ public void setImportFileParser(final ImportFileParser importFileParser) {
+
+    // expands or shrinks the import pane based on if there is a file present
+    // within the imported files list
+    void openConfigPane(boolean b) {
+        importPane.expandPane(b);
+    }
+
+    // enables or disables the import button based on if there is a file present
+    // within the imported files list
+    void enableButton(boolean b) {
```
parameters final

arcturus2 reviewed (approved):
@Atlas139mkm some merge conflicts have come up so please take a look but once @antares1470's feedback is addressed I am happy for this to be merged in.

antares1470 commented on CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/whatsnew.txt:
```diff
@@ -1,6 +1,9 @@
 == 3030-12-31 Getting Started
 <p>If you're new to Constellation, read the <a href="" helpId="au.gov.asd.tac.constellation.functionality.gettingstarted">getting started guide</a>.</p>
 
+== 2021-09-06 Improved import view UI guidance of intended flow
```
I think I would simplify the title here to something like 'Import From File'. Keeps the title simple and then the details can be put in the description

antares1470 commented on CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/whatsnew.txt:
```diff
@@ -1,6 +1,9 @@
 <p>If you're new to Constellation, read the <a href="" helpId="au.gov.asd.tac.constellation.functionality.gettingstarted">getting started guide</a>.</p>
 
+== 2021-09-06 Improved import view UI guidance of intended flow
+<p>In the import view, the configuration pane starts shrunk and the import button starts disabled. They will expand/be enabled respectively once a file is present the files list. The opposite will occur when the the list is empty.</p>
```
Import from File rather than import view. I think I would also keep the description more concise. Something like "Updated the UI to only enable and expand sections once a file has added"

Atlas139mkm commented:
Alright I have made the requested changes. I going to work on enabling the import button on the database connection tomorrow morning. I don't want to cause issues for those that connect via database.

Atlas139mkm commented:
The import button should enable and the config pane should expand when data is present within the table. This should work when a query has been successful and was able to populate the table with data.

antares1470 commented on CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/jdbc/JDBCImportController.java:
```diff
@@ -193,7 +193,20 @@ protected void updateSampleData() {
 
         if (configurationPane != null) {
             configurationPane.setSampleData(currentColumns, currentData);
+            openConfigPane(!currentColumns[0].isBlank());
+            enableButton(currentColumns[0].isBlank());
```
Looks like further up the function there is the possibility of `currentColumns` being length 0, in which case this will throw an `ArrayOutOfBoundsException` so you'll probably want to add in a boolean condition for that. I would also switch to using `StringUtils.isBlank()` and `StringUtils.isNotBlank()` is case the value of `currentColumns[0]` is null

Atlas139mkm commented:
I renamed enable button to disable button to be more inline with what it actually does. I also changed the function to check for the current data not just the columns. I also added a else statement that just would shrink and disable the elements if there is ever a case where it could occur.

arcturus2 reviewed (approved):
Tested it and it works well.

antares1470 commented on CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/jdbc/JDBCImportController.java:
```diff
@@ -193,20 +193,26 @@ protected void updateSampleData() {
-            openConfigPane(!currentColumns[0].isBlank());
-            enableButton(currentColumns[0].isBlank());
+
+            // If the current columns lenght isn't 0, expand the config pane
+            // and enable the import button
+            if (!currentData.isEmpty()) {
+                openConfigPane(StringUtils.isNotBlank(currentData.get(0).toString()));
+                disableButton(StringUtils.isBlank(currentData.get(0).toString()));
+            } else {
+                openConfigPane(false);
+                disableButton(true);
+            }
```
Given `currentData` is a `List<String[]>`, the `toString()` of an array won't ever be blank so effectively the code could be simplified to 
```
openConfigPane(!currentData.isEmpty())
disableButton(currentData.isEmpty())
```

antares1470 replied:
Should probably also change the comment to refer to `currentData` rather than `currentColumns`
